packages/framework/fluid-static/src/rootDataObject.ts

```diff
@@ -165,10 +170,37 @@ const rootDataStoreId = "rootDOId";
  */
 export function createDOProviderContainerRuntimeFactory(props: {
 	schema: ContainerSchema;
+	compatMode: CompatMode;
 }): IRuntimeFactory {
-	return new DOProviderContainerRuntimeFactory(props.schema);
+	return new DOProviderContainerRuntimeFactory(props.schema, props.compatMode);
 }
 
+const compatModeRuntimeOptions: Record<CompatMode, IContainerRuntimeOptions> = {
+	"1": {
+		// 1.x clients are compatible with TurnBased flushing, but here we elect to remain on Immediate flush mode
+		// as a work-around for inability to send batches larger than 1Mb. Immediate flushing keeps batches smaller as
+		// fewer messages will be included per flush.
+		flushMode: FlushMode.Immediate,
+		// Op compression is on by default but introduces a new type of op which is not compatible with 1.x clients.
+		compressionOptions: {
+			minimumBatchSizeInBytes: Number.POSITIVE_INFINITY, // disabled
+			compressionAlgorithm: CompressionAlgorithms.lz4,
+		},
+		// TODO: Include explicit disables for things that are currently off-by-default?
```

Comment:
FYI - I merged https://github.com/microsoft/FluidFramework/pull/20604 that impacts choices here.
Yes, I'd explicitly mention all the settings here, as they should not change over time (though it's hard to enforce that for future settings that are not known at this time).

Ideally, we have tests that help us finding non-compat settings in the future, and thus force adding new things here (for example, such test should break once https://github.com/microsoft/FluidFramework/pull/20604 is picked up).

Might be useful reference point: [filterRuntimeOptionsForVersion()](https://github.com/microsoft/FluidFramework/blob/a47c948ffde16db08a9a15093982f0f81b9f7cd3/packages/test/test-version-utils/src/compatUtils.ts#L65)